src/utility/serializable.hpp

```
@@ -0,0 +1,169 @@
+        constexpr auto N = std::tuple_size_v<Metas>;
+        return make_serializable_impl(metas, std::make_index_sequence<N / 2> {});
+    }
+
+    template <typename T>
+    auto serialize(this T& self, std::string_view prefix, const auto& source) noexcept
+        -> std::expected<void, std::string> {
+
+        static_assert(
+            details::serializable_metas_trait<T>, "Serializable T must has valid metas tuple");
+
+        auto s = self.make_serializable(self.metas);
```

Clang (14.0.6)

[warning] variable name 's' is too short, expected at least 3 characters

(readability-identifier-length)